modules/reactor/test/tests/meshgenerators/tri_pin_hex_assemby_generator/tests

```diff
@@ -0,0 +1,209 @@
+[Tests]
+  design = 'meshgenerators/TriPinHexAssemblyGenerator.md'
+  issues = '#20949'
+  [tri_pin]
+    type = 'Exodiff'
+    input = 'tri_pin.i'
+    exodiff = 'tri_pin_out.e'
+    cli_args = 'Mesh/assm_up/ring_block_names="ring_11 ring_12 ring_13;ring_21 ring_22; "
+                Mesh/assm_up/background_block_names="bg_1 bg_2"
+                Mesh/assm_up/pin_id_name=pin_id
+                Mesh/assm_up/pin_id_values="0 1 2"'
+    requirement ='The system shall generate a tri-pin assembly mesh.'
+    recover = false
+  []
+  [tri_pin_apothem]
+    type = 'Exodiff'
+    input = 'tri_pin.i'
+    exodiff = 'tri_pin_out.e'
+    cli_args = 'Mesh/assm_up/ring_block_names="ring_11 ring_12 ring_13;ring_21 ring_22; "
+                Mesh/assm_up/hexagon_size=20
+                Mesh/assm_up/hexagon_size_style=apothem
+                Mesh/assm_up/pin_id_name=pin_id
+                Mesh/assm_up/pin_id_values="0 1 2"'
+    requirement ='The system shall generate a tri-pin assembly mesh with apothem style size.'
+    recover = false
+  []
+  [tri_pin_dup]
+    type = 'Exodiff'
+    input = 'tri_pin_dup.i'
+    exodiff = 'tri_pin_dup_in.e'
+    cli_args = '--mesh-only'
+    requirement ='The system shall generate a tri-pin assembly mesh with three identical pins.'
+    recover = false
+  []
+  [dummy_core]
+    type = 'Exodiff'
+    input = 'dummy_core.i'
+    exodiff = 'dummy_core_in.e'
+    cli_args = '--mesh-only'
+    requirement ='The system shall generate a dummy core mesh.'
+    recover = false
+    #heavy = true
+  []
+  [err_ring_radii_size]
+    type = RunException
+    input = 'tri_pin.i'
+    cli_args = '--mesh-only tri_pin_err.e
+               Mesh/assm_up/ring_radii="7 8;5 6"'
+    expect_err = 'This parameter must have a size of one or three if provided.'
+    requirement = 'The system shall thow an error if ring_radii does not have appropriate size.'
```

Comment:
```suggestion
    requirement = 'The system shall throw an error if the number of rings radii specified does not match either one or the number of pins'
```
fix others too pls
When possible it s better to have a sentence that reads on its own, not referring to the parameter name btw